src/Node/Expression/Binary/EndsWithBinary.php

```diff
@@ -24,7 +24,7 @@ public function compile(Compiler $compiler): void
             ->subcompile($this->getNode('left'))
             ->raw(sprintf(') && is_string($%s = ', $right))
```

Comment:
An additional optimization could be to remove the `is_string($right = <right>)` when the `right` expression is a constant string. And using string mode, it should throw an exception if it is a constant that is not a string.